shiny/ui/_theme.py

```diff
@@ -0,0 +1,513 @@
+from __future__ import annotations
+
+import copy
+import os
+import pathlib
+import tempfile
+from textwrap import dedent
+from typing import (
+    Any,
+    Callable,
+    Iterable,
+    Literal,
+    Optional,
+    Sequence,
+    TypeVar,
+    Union,
+    cast,
+)
+
+from htmltools import HTMLDependency
+from packaging.version import Version
+
+from .._docstring import no_example
+from .._typing_extensions import NotRequired, TypedDict
+from .._versions import bootstrap
+from ._theme_presets import (
+    ShinyThemePreset,
+    shiny_theme_presets,
+    shiny_theme_presets_bundled,
+)
+from ._utils import path_pkg_www
+
+T = TypeVar("T", bound="Theme")
+
+
+SassImporterReturnValue = Union["tuple[str]", "tuple[str, str]", "tuple[str, str, str]"]
+SassImporterFunction = Union[
+    Callable[[str], SassImporterReturnValue],
+    Callable[[str, str], SassImporterReturnValue],
+]
+
+
+class SassCompileArgs(TypedDict):
```

Comment:
IIRC, `total=False` makes it so you don't need the `NotRequired`?

```suggestion
class SassCompileArgs(TypedDict, total=False):
```

Reply:
IIUC, `total=False` means that dictionaries of type `SassCompileArgs` can include keys not listed in the definition. Here, we need to use `NotRequired[]` to indicate that the dictionary doesn't need to include `output_style`, for example, and `NotRequired[]` is preferred to `Optional[]` because in the latter we'd need to provide a default value.